### PR TITLE
Fix for #183, #193

### PR DESCRIPTION
**Pull request by piyushroshan**
Fix for 
#183 Organizing rule matches
#193 Chain rules handling

> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Note**: _that go.mod and go.sum can only be modified for tested dependency updates or justified new features._

**Make sure that you've checked the boxes below before you submit PR:**

- [x] My code includes positive and negative tests.
- [x] I have an appropriate description with correct grammar.
- [x] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

File: rule_match.go [hunk 1]
```diff
@@ -58,23 +62,24 @@ type MatchedRule struct {
 	// Client IP address
 	ClientIPAddress string
 	// A slice of matched variables
-	MatchedData MatchData
```
Comment (jptosso) [inline rule_match.go:61]: This line must be marked as deprecated but not removed until v3

**Comment by fzipi** [comment]
Hi @piyushroshan,

Thanks for your submission. I'm sorry to say this, but can you send this in three different PRs, one targeting each issue? It will make reviewing easier, and even If we merge this one it is going to be very difficult to revert.

File: testing/profile.go [hunk 2]
```diff
@@ -60,8 +60,8 @@ type Profile struct {
 type expectedOutput struct {
 	Headers           map[string]string `yaml:"headers,omitempty"`
 	Data              interface{}       `yaml:"data,omitempty"` // Accepts array or string
-	LogContains       string            `yaml:"log_contains,omitempty"`
-	NoLogContains     string            `yaml:"no_log_contains,omitempty"`
+	LogContains       []string          `yaml:"log_contains,omitempty"`
```
Comment (jptosso) [inline testing/profile.go:63]: for compatibility maybe we could use interface{} and when processing the variable we can switch between string and []string

**Comment by jptosso** [comment]
I think this was already fixed  in another PR, closing

**Pull request closed without merging**